source/Cute/Commands/Content/ContentJoinCommand.cs

```diff
@@ -0,0 +1,165 @@
+
+        Action<QueryBuilder<JObject>>? queryConfigTarget =
+            string.IsNullOrEmpty(settings.Source2EntryId)
+            ? null
+            : b => b.FieldEquals($"fields.{targetField2.Id}.sys.id", settings.Source2EntryId);
+
+        Action<QueryBuilder<JObject>>? queryConfigSource2 =
+            string.IsNullOrEmpty(settings.Source2EntryId)
+            ? null
+            : b => b.FieldEquals("sys.id", settings.Source2EntryId);
+
+        var targetData = ContentfulEntryEnumerator.DeliveryEntries(ContentfulClient, joinEntry.TargetContentType, targetContentType.DisplayField, queryConfigurator: queryConfigTarget)
+            .ToBlockingEnumerable()
+            .ToDictionary(e => e.Entry["key"]!, e => new { Key = e.Entry["key"], Title = e.Entry["title"], Name = e.Entry["name"] });
+
+        var source1Data = ContentfulEntryEnumerator.DeliveryEntries<JObject>(ContentfulClient, joinEntry.SourceContentType1, source1ContentType.DisplayField)
+            .ToBlockingEnumerable()
+            .Where(e => source1AllKeys || source1Keys.Contains(e.Entry["key"]?.Value<string>()));
+
+        var source2Data = ContentfulEntryEnumerator.DeliveryEntries(ContentfulClient, joinEntry.SourceContentType2, source2ContentType.DisplayField, queryConfigurator: queryConfigSource2)
+            .ToBlockingEnumerable()
+            .Where(e => source2AllKeys || source2Keys.Contains(e.Entry["key"]?.Value<string>()));
+
+        // Start Join
+
+        var limit = 0;
+
+        foreach (var (entry1, _) in source1Data)
+        {
+            foreach (var (entry2, _) in source2Data)
+            {
+                var joinKey = $"{entry1["key"]?.Value<string>()}.{entry2["key"]?.Value<string>()}";
+                var joinTitle = $"{entry1["title"]?.Value<string>()} | {entry2["title"]?.Value<string>()}";
+                var joinName = $"{entry2["name"]?.Value<string>()}";
+
+                if (targetData.ContainsKey(joinKey)) continue;
+
+                var newFlatEntry = targetSerializer.CreateNewFlatEntry();
+                newFlatEntry[$"key.{defaultLocale}"] = joinKey;
+                newFlatEntry[$"title.{defaultLocale}"] = joinTitle;
+                newFlatEntry[$"name.{defaultLocale}"] = joinName;
+                newFlatEntry[$"{targetField1.Id}.{defaultLocale}"] = entry1["$id"];
+                newFlatEntry[$"{targetField2.Id}.{defaultLocale}"] = entry2["$id"];
+
+                var newEntry = targetSerializer.DeserializeEntry(newFlatEntry);
+
+                _console.WriteNormal("Creating {joinTargetContentTypeId} '{joinKey}' - '{joinTitle}'", joinEntry.TargetContentType, joinKey, joinTitle);
+
+                await UpdateAndPublishEntry(newEntry, joinEntry.TargetContentType);
+
+                if (++limit >= settings.Limit)
+                {
+                    break;
+                }
+            }
+        }
+
+        return 0;
+    }
+
+    private async Task UpdateAndPublishEntry(Entry<JObject> newEntry, string contentType)
```

Comment:
Remove these (By the way they aren't rate limited and we should wrap all Contentful calls in RateLimiter). Use Bulk action pattern above as this is far more performant, handles updates and rate limits and displays progress consistently